WordPress/WordPressApi/WordPressComRestApi.swift

```diff
@@ -227,6 +230,7 @@ final class WordPressComRestAPIResponseSerializer: AFJSONResponseSerializer
         let extraStatusCodes = NSMutableIndexSet(indexSet: self.acceptableStatusCodes!)
         extraStatusCodes.addIndexesInRange(NSRange(400...500))
         self.acceptableStatusCodes = extraStatusCodes
+        self.readingOptions = [.AllowFragments]
```

Comment:
Funny, the spec you linked actually states that

> A JSON text is a serialized object or array.
>      JSON-text = object / array

There a new specification from 2014 in [RFC 7159](https://tools.ietf.org/html/rfc7159) that changes that

> A JSON text is a serialized value.  Note that certain previous
   specifications of JSON constrained a JSON text to be an object or an
   array.  Implementations that generate only objects or arrays where a
   JSON text is called for will be interoperable in the sense that all
   implementations will accept these as conforming JSON texts.

The change is fairly recent, and probably why `.AllowFragments` is an option and not the default behavior.

I don't recall when the wp.com REST API was introduced, but our first usage of REST in the app predates the revised JSON spec by over a year.

And even if it's valid JSON, I think our API will always use an object as a response. Even something like [removing a user](https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/users/%24user_ID/delete/) returns a `{success: true}` instead of a value. That's what I meant by "assuming a dictionary".

The `is-available/email` endpoint is not really part of the REST API (it's not under `/rest/`), and it shouldn't change the behavior for the rest of the endpoints.

Overall, I just worry that we're open the door for unexpected bugs just to handle a rare temporary exception that should be fixed in the backend.